src/modules/firestore/FieldValue.js

```diff
@@ -2,16 +2,39 @@
  * @flow
  * FieldValue representation wrapper
  */
+import AnyJs from './utils/any';
 
 export default class FieldValue {
+  _elements: AnyJs[];
+
+  get elements(): AnyJs[] {
+    return this._elements;
+  }
+
+  set elements(elements: AnyJs[]) {
+    this._elements = elements;
+  }
+
   static delete(): FieldValue {
     return DELETE_FIELD_VALUE;
   }
 
   static serverTimestamp(): FieldValue {
     return SERVER_TIMESTAMP_FIELD_VALUE;
   }
+
+  static arrayUnion(...elements: AnyJs[]) {
+    ARRAY_UNION_FIELD_VALUE.elements = elements;
```

Comment:
A small bug here; this should return a new instance of FieldValue and not be a singleton like the others.

If you add a test with 2x `FieldValue`.`arrayUnion` or 2x `arrayRemove` (with different values) then it'll fail as only the 2nd `FieldValue` will overwrite the elements of the first `FieldValue`, if that makes sense?

Changing this will the `if (value === ARRAY_UNION_FIELD_VALUE) {` checks - so they'd need changing also.